src/dotnet/Chat.UI.Blazor/Components/ChatNavbarItem.razor

```diff
@@ -0,0 +1,75 @@
+@using BlazorContextMenu
+@inherits ComputedStateComponent<ChatNavbarItem.Model>
+@{
+    var m = State.LatestNonErrorValue;
+}
+
+<ContextMenuTrigger MouseButtonTrigger="MouseButtonTrigger.Right" MenuId="@ChatMenu.MenuId" Data="@Chat" CssClass="flex-x items-center gap-x-2">
+    <NavbarItem Url="@($"/chat/{Chat.Id}")" IsActive="@IsActive">
+        <ChildContent>
+            <div class="flex-x items-center gap-x-2">
+                <ChatIcon Title="@Chat.Title" Picture="@Chat.Picture" Size="SquareSize.Size10" />
+                <span class="text-ellipsis whitespace-nowrap overflow-hidden">@Chat.Title</span>
+            </div>
+        </ChildContent>
+        <Ending>
+            @if (IsListeningPanel) {
+                <ChatListListenToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                @if (m.CanRecord) {
+                    <ChatListRecordingToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                }
+                <ListeningChatRemoveButton ChatId="@Chat.Id"/>
+            } else {
+                <ChatListListenToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                @if (m.CanRecord) {
+                    <ChatListRecordingToggle ChatId="@Chat.Id" Class="chat-menu-btn blackout"/>
+                }
+                <ContextMenuToggleTrigger MenuId="@ChatMenu.MenuId" Data="@Chat">
+                    <div class="flex">
+                        <Button Class="btn-round chat-menu-btn blackout">
+                            <svg class="w-6 h-6 fill-current" xlmns="http://www.w3.org/2000/svg" viewBox="-1 -1 26 26">
+                                <path fill-rule="evenodd" clip-rule="evenodd" d="M12 7C13.104 7 14 6.104 14 5C14 3.896 13.104 3 12 3C10.896 3 10 3.896 10 5C10 6.104 10.896 7 12 7ZM12 10C10.896 10 10 10.896 10 12C10 13.104 10.896 14 12 14C13.104 14 14 13.104 14 12C14 10.896 13.104 10 12 10ZM10 19C10 17.896 10.896 17 12 17C13.104 17 14 17.896 14 19C14 20.104 13.104 21 12 21C10.896 21 10 20.104 10 19Z"/>
+                            </svg>
+                        </Button>
+                        <div class="absolute -right-4">
+                            <div class="placement -mt-2">
+                            </div>
+                        </div>
+                    </div>
+                </ContextMenuToggleTrigger>
+
+                <UnreadMessageCounter ChatId="@Chat.Id"/>
+            }
+        </Ending>
+    </NavbarItem>
+</ContextMenuTrigger>
+
+@code {
+    [Inject] private IChats Chats { get; init; } = null!;
+    [Inject] private Session Session { get; init; } = null!;
+
+    [Parameter] public Chat Chat { get; init; } = null!;
+    [Parameter] public bool IsActive { get; init; }
+    [Parameter] public bool IsListeningPanel { get; init; }
+
+    protected override ComputedState<Model>.Options GetStateOptions() {
+        return new ComputedState<Model>.Options() {
```

Comment:
return new ()

Reply:
This would not work, apparently, due to generic type.